agent/stats/queue_test.go

```diff
@@ -309,3 +310,55 @@ func TestCpuStatsSetNotSetToInfinity(t *testing.T) {
 		t.Errorf("Computed cpuStatsSet.SampleCount (%d) != expected value (%d)", sampleCount, 1)
 	}
 }
+
+func TestResetThresholdElapsed(t *testing.T) {
+	// create a queue
+	queueLength := 3
+	queue := NewQueue(queueLength)
+
+	queue.Reset()
+
+	thresholdElapsed := queue.resetThresholdElapsed(2 * time.Millisecond)
+	assert.False(t, thresholdElapsed, "Queue is NOT expected to elapsed the threshold right after reset")
```

Comment:
I don't think the phrasing is correct here. The Queue does not elapse any threshold as it's not an entity that is time-bound. I think the better message here is "Queue reset threshold is not expected to elapse right after reset"